src/components/entity-highlighter/AddEntityInput.js

```diff
@@ -0,0 +1,25 @@
+import React, { useState } from "react";
+import { styles } from "./styles";
+
+export const AddEntityInput = ({selectionStart, selectionEnd, entities, setEntities}) => {
+  const [inputText, setInputText] = useState('');
```

Comment:
As the `inputText` is only needed by our `AddEntityInput` component, we can have this piece of state "deep" in the component tree, which guarantees no re-renders of components that don't care about this value.